src/lib/systemd_journal/journalctl.rb

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2014 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+module SystemdJournal
+  # Wrapper for journalctl invocation
+  class Journalctl
+
+    BASH_SCR_PATH = Yast::Path.new(".target.bash_output")
+    COMMAND = "LANG=C journalctl"
+    TIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+    attr_reader :options, :matches
+
+    # @param options [Hash] The keys are options of the journalctl command like
+    #   "boot" or "no-pager" (the long option name should be used). The values
+    #   are the corresponding values for the option or nil for options without
+    #   an expected value. For options that can be specified multiple times per
+    #   command, an array can be used.
+    # @param options [Array] list of journalctl matches
+    #
+    # @example Kernel messages in the last minute for a given device and units
+    #   Journalctl.new(
+    #     {
+    #       "since" => Time.now - 60,
+    #       "unit"  => ["local-fs.target", "swap.target"],
+    #       "dmesg" => nil
+    #     },
+    #     ["/dev/sda"]
+    #   )
+    def initialize(options = {}, matches = [])
+      @options = options
+      @matches = matches
+    end
+
+    # Full journalctl command
+    def command
+      "#{COMMAND} #{options_string} #{matches_string}".strip.squeeze(" ")
+    end
+
+    # Output resulting of executing the command
+    def output
+      return @output if @output
```

Comment:
why caching output? this looks strange, as cached result can be invalid after some time. Be aware that stuff that are not always same need to be invalidated when it start to be outdated or invalid.